core/src/main/java/com/linecorp/armeria/server/Server.java

```diff
@@ -574,14 +575,19 @@ private ChannelFuture doStart(ServerPort port) {
         }
 
         private void setupServerMetrics() {
-            final MeterRegistry meterRegistry = config().meterRegistry();
+            final MeterRegistry meterRegistry = config.meterRegistry();
+            final ServerMetrics serverMetrics = config.serverMetrics();
             final GracefulShutdownSupport gracefulShutdownSupport = this.gracefulShutdownSupport;
             assert gracefulShutdownSupport != null;
 
             meterRegistry.gauge("armeria.server.pending.responses", gracefulShutdownSupport,
                                 GracefulShutdownSupport::pendingResponses);
-            meterRegistry.gauge("armeria.server.connections", connectionLimitingHandler,
-                                ConnectionLimitingHandler::numConnections);
+            meterRegistry.gauge("armeria.server.connections", serverMetrics,
+                                ServerMetrics::activeConnections);
+            meterRegistry.gauge("armeria.server.pending.requests", serverMetrics,
+                                ServerMetrics::pendingRequests);
+            meterRegistry.gauge("armeria.server.active.requests", serverMetrics,
+                                ServerMetrics::activeRequests);
```

Comment:
Should we tag `SessionProtocol` to the metric?

Reply:
All addressed. PTAL. 😉